### PR TITLE
Booster implementation

### PR DESCRIPTION
**Pull request by octotentakel**

### CODE REVIEW

**Comment by TehMessiah** [comment]
Could one of the administrators verify this pull-request?

**Comment by BeeHiveJava** [comment]
test this please

File: Soccar [Physics]/src/nl/soccar/physics/models/WheelPhysics.java [hunk 1]
```diff
@@ -94,14 +95,14 @@ public WheelPhysics(float relPosX, float relPosY, float width, float height, boo
...
-     * Apply force on the wheel based on the power of the car and the desired speed.
+     * Apply force on the wheel based on the power of the carPhysics and the desired speed.
      */
     public void updateDrive() {
 
         Vec2 currentForwardNormal = body.getWorldVector(new Vec2(0, 1));
 
         float currentSpeed = Vec2.dot(getForwardVelocity(), currentForwardNormal);
-        float force = PhysicsConstants.CAR_POWER * 10;
+        float force = power * 10;
```
Comment (TehMessiah) [inline Soccar [Physics]/src/nl/soccar/physics/models/WheelPhysics.java]: CRITICAL Cast one of the operands of this multiplication operation to a "float". [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS2184)

File: Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java [hunk 2]
```diff
@@ -125,6 +131,33 @@ private void updateSteerAngle() {
...
+            boostActive = true;
+        }
+
+        // Disable boost when the throttle action is not boost or boost amount is 0
+        if (car.getThrottleAction() != ThrottleAction.BOOST || car.getBoostAmount() == 0) {
+            boostActive = false;
+        }
+
+        // The trail has to remove itself when it's longer than max length or boost is not active.
+        if (trail.size() > PhysicsConstants.CAR_BOOST_TRAIL_SIZE || trail.size() > 0 && !boostActive) {
```
Comment (TehMessiah) [inline Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1155)

File: Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java [hunk 3]
```diff
@@ -125,6 +131,33 @@ private void updateSteerAngle() {
         }
     }
 
+    private void updateBoost() {
+
+        // Only allow boosting when the trail is gone.
+        if (car.getThrottleAction() == ThrottleAction.BOOST && trail.size() == 0) {
```
Comment (TehMessiah) [inline Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1155)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 4 issues
* CRITICAL 1 critical
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Review by Ranray — APPROVED** [review]

**Pull request merged**